src/debugger/scalaDebugger.ts

```diff
@@ -64,7 +75,7 @@ async function runMain(main: ExtendedScalaRunMain): Promise<boolean> {
       workspace.workspaceFolders[0],
       "Scala run",
       "Metals",
-      new ShellExecution(main.data.shellCommand, { env })
+      new ShellExecution(main.data.shellCommand, shellOptions(env))
```

Comment:
some sugar with `const { environmentVariables, shellCommand } = main.data;` ;)
```ts
  const { environmentVariables, shellCommand } = main.data;
  if (workspace.workspaceFolders) {
    const env = environmentVariables.reduce<Record<string, string>>(
      (acc, envKeyValue) => {
        const [key, value] = envKeyValue.split("=");
        return { ...acc, [key]: value };
      },
      {}
    );

    const shellOptions = { ...platformSpecificOptions(), env };

    const task = new Task(
      { type: "scala", task: "run" },
      workspace.workspaceFolders[0],
      "Scala run",
      "Metals",
      new ShellExecution(shellCommand, shellOptions)
    );
```

Reply:
this is some crazy magic, changed!